tests/nonlinear_channel_flow_velocities_iterated_IMPES/screen-output

```diff
@@ -7,268 +9,1600 @@ Number of degrees of freedom: 3,556 (2,178+289+1,089)
 
 
    Skipping temperature solve because RHS is zero.
    Rebuilding Stokes preconditioner...
-   Solving Stokes system... 7+0 iterations.
-      Relative nonlinear residuals: 0, 4.07482e-13
-      Total relative residual after nonlinear iteration 33: 4.07482e-13
+   Solving Stokes system... 0+0 iterations.
+      Relative nonlinear residuals: 0, 9.92785e-09
+      Total relative residual after nonlinear iteration 8: 9.92785e-09
 
 
    Skipping temperature solve because RHS is zero.
    Rebuilding Stokes preconditioner...
-   Solving Stokes system... 6+0 iterations.
-      Relative nonlinear residuals: 0, 2.64737e-13
-      Total relative residual after nonlinear iteration 34: 2.64737e-13
+   Solving Stokes system... 0+0 iterations.
+      Relative nonlinear residuals: 0, 9.92785e-09
+      Total relative residual after nonlinear iteration 9: 9.92785e-09
 
 
    Skipping temperature solve because RHS is zero.
    Rebuilding Stokes preconditioner...
-   Solving Stokes system... 5+0 iterations.
-      Relative nonlinear residuals: 0, 1.72179e-13
-      Total relative residual after nonlinear iteration 35: 1.72179e-13
+   Solving Stokes system... 0+0 iterations.
+      Relative nonlinear residuals: 0, 9.92785e-09
+      Total relative residual after nonlinear iteration 10: 9.92785e-09
 
 
    Skipping temperature solve because RHS is zero.
    Rebuilding Stokes preconditioner...
-   Solving Stokes system... 4+0 iterations.
-      Relative nonlinear residuals: 0, 1.12207e-13
-      Total relative residual after nonlinear iteration 36: 1.12207e-13
+   Solving Stokes system... 0+0 iterations.
+      Relative nonlinear residuals: 0, 9.92785e-09
+      Total relative residual after nonlinear iteration 11: 9.92785e-09
+
+
+   Skipping temperature solve because RHS is zero.
+   Rebuilding Stokes preconditioner...
+   Solving Stokes system... 0+0 iterations.
+      Relative nonlinear residuals: 0, 9.92785e-09
+      Total relative residual after nonlinear iteration 12: 9.92785e-09
+
+
+   Skipping temperature solve because RHS is zero.
+   Rebuilding Stokes preconditioner...
+   Solving Stokes system... 0+0 iterations.
+      Relative nonlinear residuals: 0, 9.92785e-09
+      Total relative residual after nonlinear iteration 13: 9.92785e-09
+
+
+   Skipping temperature solve because RHS is zero.
+   Rebuilding Stokes preconditioner...
+   Solving Stokes system... 0+0 iterations.
+      Relative nonlinear residuals: 0, 9.92785e-09
+      Total relative residual after nonlinear iteration 14: 9.92785e-09
+
+
+   Skipping temperature solve because RHS is zero.
+   Rebuilding Stokes preconditioner...
+   Solving Stokes system... 0+0 iterations.
+      Relative nonlinear residuals: 0, 9.92785e-09
+      Total relative residual after nonlinear iteration 15: 9.92785e-09
+
+
+   Skipping temperature solve because RHS is zero.
+   Rebuilding Stokes preconditioner...
+   Solving Stokes system... 0+0 iterations.
```

Comment:
Is there anything we can do about this case? The linear solver does not solve anything, because it thinks it has converged, but the nonlinear residual is not yet reached, so it keeps iterating.

Reply:
ah, I havn't seen this in a long time (didn't pay much attention to the convergence of the impes, just made sure the result was the same a the analytical result). This goes back to #302, where back then I couldn't find a case any more to reproduce the problem.